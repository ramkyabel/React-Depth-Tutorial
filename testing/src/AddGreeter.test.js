import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddGreeter from './AddGreeter';

describe (AddGreeter, () => {
	const mockAddGreeting = jest.fn ();
	const mockRemoveGreeting = jest.fn ();
	const component = shallow (
		<AddGreeter addGreeting={mockAddGreeting} 
								removeGreeting={mockRemoveGreeting} />);

	it ('renders and matches our snapshot',()=> {
		const component = renderer.create (
			<AddGreeter />
		);
		const tree = component.toJSON ();
		expect (tree).toMatchSnapshot ();
	});

 it ('contains the form', () => {
 		expect (component.find ('div')).toHaveLength (1);
 		expect (component.find ('input')).toHaveLength (1);
 		expect (component.find ('button')).toHaveLength (2);
 });

 it ('calls the passed in addGreeting and removeGreeting functions' +
 'when add and remove buttons are clicked', () => {
 	component.find ('button').at(0).simulate ('click');
 	expect (mockAddGreeting).toBeCalled ();
	component.find ('button').at(1).simulate ('click');
 	expect (mockRemoveGreeting).toBeCalled ();
 });

 it ('updates the form when keys are pressed', () => {
 	const updateKey = 'bar';
 	component.instance ().handleUpdate ({ target: { value: updateKey}});
 	expect (component.state ('greetingName')).toEqual (updateKey);
 });

 it ('blanks out the greetingName when the button is clicked', () => {
 	const updateKey = 'bar';
 	component.instance ().handleUpdate ( {target: { value : updateKey}});
 	expect (component.state ('greetingName')).toEqual (updateKey);
 	component.find ('button').at (0).simulate ('click');
 	expect (component.state ('greetingName')).toEqual ('');
 });
});