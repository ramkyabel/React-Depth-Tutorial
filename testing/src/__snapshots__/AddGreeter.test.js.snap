// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class AddGreeter extends _react.Component {
	constructor(props) {
		super(props);
		this.state = { greetingName: '' };
		this.handleUpdate = this.handleUpdate.bind(this);
		this.addGreeting = this.addGreeting.bind(this);
		this.removeGreeting = this.removeGreeting.bind(this);
	}
	addGreeting() {
		this.props.addGreeting(this.state.greetingName);
		this.setState({ greetingName: '' });
	}
	removeGreeting() {
		this.props.removeGreeting(this.state.greetingName);
		this.setState({ greetingName: '' });
	}
	handleUpdate(event) {
		this.setState({ greetingName: event.target.value });
	}
	render() {
		return (
			_react2.default.createElement('div', { className: 'AddGreeter', __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: this },
				_react2.default.createElement('input', {
					type: 'text',
					onChange: this.handleUpdate,
					value: this.state.greetingName, __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this }), '\\xA0\\xA0',



				_react2.default.createElement('button', { className: 'add', onClick: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this }, ' Add '),
				_react2.default.createElement('button', { className: 'remove', onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this }, 'Remove')));


	}} renders and matches our snapshot 1`] = `
<div
  className="AddGreeter"
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
    
  <button
    className="add"
    onClick={[Function]}
  >
     Add 
  </button>
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
